File: sensorNode/sensor_node.py
import requests
import random
import time
from time import sleep

SERVER1_URL='http://10.2.1.27:8080/regional_server'
SERVER2_URL='http://10.2.1.34:8080/regional_server'


def sensor_send(url=SERVER1_URL, s_id='TS0N', d_type='temperature'):
    t = time.time()
    timestamp = int(t*1000)
    if d_type == 'temperature':
        reading = random.randint(10, 45)
    elif d_type == 'humidity':
        reading = random.randint(60, 99)
    string_to_send = 'sensor_id='+s_id+'&sensor_data='+str(reading)+'&sensor_data_type='+d_type+'&time='+str(timestamp)
    r = requests.post(url, data = string_to_send)
    print("Sensor: ",s_id," Sent Reading: ", str(reading))
    print("Regional Server Reply: ", r.text)
    
def ts01_send():
    sensor_send(SERVER1_URL, 'TS01', 'temperature')
    
def hs01_send():
    sensor_send(SERVER1_URL, 'HS01', 'humidity')

def ta01_send():
    sensor_send(SERVER1_URL, 'TA01', 'temperature')

def ha01_send():
    sensor_send(SERVER1_URL, 'HA01', 'humidity')

def ts02_send():
    sensor_send(SERVER2_URL, 'TS02', 'temperature')

def hs02_send():
    sensor_send(SERVER2_URL, 'HS02', 'humidity')

def tb02_send():
    sensor_send(SERVER2_URL, 'TB02', 'temperature')

def hb02_send():
    sensor_send(SERVER2_URL, 'HB02', 'humidity')

def run_test(mode = 1, runs = 6000):
    if mode == 1:
        for i in range (runs):
            ts01_send()
            sleep(0.25)
            hs01_send()
            sleep(0.25)

    elif mode == 2:
        for i in range (runs):
            ts02_send()
            sleep(0.25)
            hs02_send()
            sleep(0.25)

    elif mode == 3:
        for i in range (runs):
            ta01_send()
            sleep(0.25)
            ha01_send()
            sleep(0.25)
            tb02_send()
            sleep(0.25)
            hb02_send()
            sleep(0.25)
    

File: sensorNode/SensorNode.java
package sensorNode;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.OutputStream;
import java.net.HttpURLConnection;
import java.net.URL;
import java.time.Instant;
import java.util.Random;

public class SensorNode {
    private static final String REGIONAL_SERVER_URL = "http://10.2.1.27:8080/regional_server";
    private static final int SEND_INTERVAL = 1000;
    public static void main(String[] args) {
        while (true) {
            try {

                sendSensorData("HS01", "humidity");
                sendSensorData("TS01", "temperature");
                Thread.sleep(SEND_INTERVAL);
            } catch (Exception e) {
                e.printStackTrace();
            }
        }
    }

    private static void sendSensorData(String sensorId, String sensorDataType) throws Exception {
        String sensorData = generateSensorData(sensorId, sensorDataType);
        System.out.println("Generated sensor data: " + sensorData);
        HttpURLConnection connection = null;

        try {
            URL url = new URL(REGIONAL_SERVER_URL);
            connection = (HttpURLConnection) url.openConnection();
            connection.setRequestMethod("POST");
            connection.setDoOutput(true);

            OutputStream os = connection.getOutputStream();
            os.write(sensorData.getBytes());
            os.flush();
            os.close();

            int responseCode = connection.getResponseCode();
            System.out.println("Response code: " + responseCode);

            BufferedReader br = new BufferedReader(new InputStreamReader(connection.getInputStream()));
            String line;
            StringBuilder response = new StringBuilder();

            while ((line = br.readLine()) != null) {
                response.append(line);
            }
            br.close();

            System.out.println("Response: " + response.toString());
        } finally {
            if (connection != null) {
                connection.disconnect();
            }
        }
    }


    private static String generateSensorData(String sensorId, String sensorDataType) {
        Random random = new Random();
        int sensorData;

        if (sensorDataType.equals("temperature")) {
            sensorData = random.nextInt(21) + 10;
        } else {
            sensorData = random.nextInt(15) + 85;
        }

        long currentTime = Instant.now().toEpochMilli();
        return String.format("sensor_id=%s&sensor_data=%d&sensor_data_type=%s&time=%d",
                sensorId, sensorData, sensorDataType, currentTime);
    }

}

